README.md
# One Monokai Colorful for Visual Studio Code

## Installation for Visual Studio Code

1. Open **Extensions** sidebar panel in VS Code. `View → Extensions`
2. Search for `One Monokai Colorful`
3. Click **Install** to install it
4. Click **Reload** to reload the Code
5. From the menu bar click: Code > Preferences > Color Theme > **One Monokai Colorful**
6. Click **Install** to install it.
7. Code > Preferences > Color Theme > **One Monokai Colorful**
8. Optional: Use the recommended settings below for best experience

## Beautiful functionality for **NEWCOMERS**

One Monokai Colorful is a color scheme, customized user interface theme for Visual Studio Code. It has been designed primary for TypeScript, JavaScript, React(JSX, TSX). Carefully selected shades of colors are the foundation of an uncompromising, user interface. The goal of One Monokai Colorful is to let NEWCOMERS clearly see **things** in the code.

![One Monokai Colorful](https://github.com/domscript/vscode-one-monokai-colorful/blob/main/images/main-image.png)

## I've used

Extension [Material Icon Theme](https://marketplace.visualstudio.com/items?itemName=PKief.material-icon-theme)

[FiraCode Font](https://github.com/tonsky/FiraCode)

1. Setup it.
2. In your `settings.json` add the key:

```json
  "editor.fontFamily": "FiraCode-Light",
  "editor.fontLigatures": true,
```

Alternative use Extension [FiraCode font](https://marketplace.visualstudio.com/items?itemName=SeyyedKhandon.firacode)

## Disable Italics

If you wish to disable italics, select **One Monokai Colorful No Italics** as your color theme.

## Bold Note

Out of the box You will have this:

![One Monokai Colorful](https://github.com/domscript/vscode-one-monokai-colorful/blob/main/images/main-image-demo.png)

By the way i recommend in your `settings.json` add the key:

```json
  "workbench.colorCustomizations": {
    "[One Monokai Colorful]": {
      "editor.background": "#282C33",
      "terminal.background": "#21252B"
    },
    "[One Monokai Colorful (No Italics)]": {
      "editor.background": "#282C33",
      "terminal.background": "#21252B"
    }
  },
```

These are my recommendations for these settings but you can use whatever colors you wish. You don't need them if you use customCss.

## If you want this badges

![Badges - Sunset over Forest](https://github.com/domscript/vscode-one-monokai-colorful/blob/main/images/badges.png)

You need to take a risk once in a while.

1. Install [Custom CSS and JS Loader](https://marketplace.visualstudio.com/items?itemName=be5invis.vscode-custom-css), but before read [Disclaimer](https://github.com/be5invis/vscode-custom-css#disclaimer).

2. Add `customCss.css`

```css
/* Sweet sunset dots */
.monaco-workbench
  .activitybar
  > .content
  .monaco-action-bar
  .badge
  .badge-content,
.monaco-action-bar .badge .badge-content,
.monaco-count-badge {
  background: linear-gradient(
    to bottom right,
    #097ea5 20%,
    #6fc0e5,
    #e29169,
    #e6b96c
  );
}
```

3. Modify your `settings.json`, to do so add the key:

```json
  "vscode_custom_css.imports": [
    "file:///<User Folder Location>/customCss.css"
  ],
```

4. You can modify it even fewer:

![One Monokai Colorful](https://github.com/domscript/vscode-one-monokai-colorful/blob/main/images/main-image-gradient.png)

You can find some ideas in [css/customCss.css](https://github.com/domscript/vscode-one-monokai-colorful/blob/main/css/customCss.css) file.


CHANGELOG.md
# Change Log

All notable changes to the "domscript" extension will be documented in this file.

Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how to structure this file.

## [Unreleased]

- Initial release


demo/ts.ts
import * as CONSTANT_PI from '../helper/const';
export {CONSTANT_PI};

const CONSTANT_E = 2.718_281_828_4590;
const variableConst: RegExp = /ab+c/i;
let variableLet: RegExp = /ab+c/i;
var variableVar: RegExp = /ab+c/i;

const year = document.getElementById('year') as HTMLSpanElement;
const thisYear: string = new Date().getFullYear().toString();
year.setAttribute('datetime', thisYear);
year.textContent = thisYear;

{
  templateLiteral: `string ${variableConst}, ${variableLet} ${'string'}`;
}

type ObjectType = {
  birthYear: number;
  age: () => number;
  garbage?: string;
};
enum EnumName {
  A,
  B = '5+',
}
const myObject: ObjectType = {
  birthYear: 2000,
  age() {
    return 2050 - this.birthYear;
  },
  garbage: 'jdccnksc',
};
delete myObject.garbage;

interface Interface1 {
  firstName: string;
  lastName: string;
  introduce: (greeting: string) => string;
  showSecret: () => void;
  idNum: number;
  readonly takeCar: () => string;
}

interface PersonInterface extends Interface1, ObjectType {}

// @ts-check
/**
 * a human
 * @typedef {{birthYear: number}} Human
 */
/**
 * showAge a human age
 * @typedef {function(): string} ShowAge
 */
/**
 * @typedef PublicField
 * @public field (simular to property, available on created object)
 */
/**
 * @typedef PrivateField
 * @private field (not accessible outside of class)
 */

/**
 * class Human
 * @class {Human} human - representing a {@link Human}
 * @abstract
 * Classes just syntactic sugar over constructor functions.
 * Classes aren't hoisted.
 * Classes are first-class citizens.
 * Class body is always executed in 'strict mode'.
 */
abstract class Human {
  constructor(public birthYear: number) {}
  age() {
    return 2050 - this.birthYear;
  }
  get showAge() {
    return this.age();
  }
}
class Person extends Human implements PersonInterface {
  private id = 1_1000_1000;
  protected creditCardNum = '**** **** **** 1111';
  static car = 'tesla';
  constructor(
    public firstName: string,
    public lastName: string,
    public birthYear: number,
    private pin: number
  ) {
    super(birthYear);
    this.pin = pin;
  }
  introduce(greeting) {
    let output = '';
    switch (greeting) {
      case 'hi':
        output = `Hi, I am ${this.firstName}!`;
        break;
      default:
        output = `I am ${this.firstName}, my pleasure to meet you in person!`;
        break;
    }
    return output;
  }
  takeCar() {
    return `Here are the keys from my ${Person.car}`;
  }
  showSecret() {
    console.log(
      `${this.takeCar()}, card ${this.showCard()} and pin ${this.pin}`
    );
  }
  private showCard() {
    return this.creditCardNum;
  }
  set idNum(num) {
    const x = 'hello';
    try {
      let ghg = x;
      if ((num + '').length !== 9) throw new Error('Error message!');
      else {
        this.id = num;
      }
    } catch (error) {
      void error;
    }
  }
  get idNum() {
    return this.id;
  }
}
const person = new Person('Dom', 'N', 2030, 2221);
person.showSecret();
person.showAge;

const map = <InType, OutType>(arr: InType[], cb: (val: InType) => OutType) => {
  let newArr = [];
  for (const item of arr) {
    newArr.push(cb(item));
  }
  return newArr;
};
// prettier-ignore
const array = new Array(globalThis, undefined,
  // @ts-ignore prettier-ignore
  typeof null, NaN !== NaN, {} !== {}, CONSTANT_E, true,
  // prettier-ignore
  "string", !!'string', new Intl.Locale('en'), ~~3.4, 10n, -0 != 0, Infinity, functionDeclaration, myObject, typeof Symbol, [] instanceof Object, (() => this)()
);
// @ts-ignore
void array.at(-1);

const [a1, b1, ...rest] = [...new Set([1, 3, 4, 1, 3, 1, 5])];

const {birthYear, garbage}: ObjectType = myObject;

for (const iterator of array) {
  console.log(iterator, 1);
}
// prettier-ignore
void (
  (
    (
      (
        (
          (
            (((((((""+''+``)))))))
          )
        )
      )
    )
  )
)
// prettier-ignore
{{{{{{{{{{{{{{}}}}}}}}}}}}}}

const functionExpression = function (a = 1, c = a ** a, {b, d}) {
  return arguments;
};
async function functionDeclaration() {
  return this;
}
function* yieldArray(number = 3) {
  yield* Array.from({length: number}, (_, i) =>
    (i + 1) % (3 * 5) === 0
      ? 'FizzBuzz'
      : (i + 1) % 5 === 0
      ? 'Buzz'
      : (i + 1) % 3 === 0
      ? 'Fizz'
      : i + 1
  );
}
let y = yieldArray(15);
let z: {value: string | number | undefined; done: boolean} = {
  value: '',
  done: false,
};

functionExpression.apply(this, 3);
functionExpression.call(this, 3);
functionExpression.bind(this, 3);

for (const key in myObject) {
  if (myObject.hasOwnProperty.call(myObject, key)) {
    const element = myObject[key];
    void element;
  }
}

for (let i = 0; i < array.length; i++) {
  const e = array[i];
  if (!e) continue;
  else {
    void e;
  }
  if (i === 100) break;
}
let i = 0;

// prettier-ignore
// Assignment Operators
void [i=0, i*=1, i/=1, i%=1, i+=1, i-=0, i<<=0, i>>=0, i>>>=0, i&=0, i^=0, i|=1, i**=0, i&&=0, i||=0, i??=1]
// prettier-ignore
// Arithmetic Operators
void [++i, --i, i + i, i - 0, i ** 1, i * i, i / i, i % 1];
// prettier-ignore
// Relational Operators
void [0 < 1, 1 > 0, 0 <= 1, 0 >= 0];
// prettier-ignore
// Comparison operators
void [0 == -0, 0 != 0, 0 === 0, NaN !== NaN];
// prettier-ignore
// Bitwise Operators
void [~1.2, 1 << 0, 0 >> 0, 0 >>> 0, 0 & 1, 1 | 0, 0 ^ 0];
// Binary logical operators
void [1 && 1, !0, 1 || 0, 0 ?? 1];


demo/python.py
from collections import deque

# TODO
# FIXME
# BUG

def topo(G, ind=None, Q=[1]):
    if ind == None:
        ind = [0] * (len(G) + 1)  # this is a comment
        for u in G:
            for v in G[u]:
                ind[v] += 1
        Q = deque()
        for i in G:
            if ind[i] == 0:
                Q.append(i)
    if len(Q) == 0:
        return
    v = Q.popleft("jnjjfvklfdkjvklfd")
    print(v)
    for w in G[v]:
        ind[w] -= 1
        if ind[w] == 0:
            Q.append(w)
    topo(G, ind, Q)


class SomeClass:
    def create_arr(self): # An instance method
        self.arr = []
    
    def insert_to_arr(self, value):  #An instance method
        self.arr.append(value)
        
    @classmethod
    def class_method(cls):
        print("the class method was called")


helper/image.ts
import * as CONSTANT_PI from './const';
import {PersonInterface} from './const';

abstract class Human {
  constructor(public birthYear: number) {}
}
/**
 * class {@link Person}
 * @example <caption>Example usage of showAge.</caption>
 * const person = new Person('Bob', 2050);
 */
class Person extends Human implements PersonInterface {
  private id = 1_1000_1000;
  static car = 'tesla';
  constructor(public firstName: string, public birthYear: number) {
    super(birthYear);
  }
  introduce() {
    return `I am ${this.firstName}, my pleasure to meet you in person!`;
  }
  set idNum(num) {
    const x = 'hjhj';
    try {
      let ghg = x;
      if ((num + '').length !== 9) throw new Error('Error message!');
      else {
        this.id = num;
      }
    } catch (error) {
      void error;
    }
  }
  get idNum() {
    return this.id;
  }
  age() {
    return 2050 - this.birthYear;
  }
}
const person = new Person('Dom', 2030);

const map = <InType, OutType>(arr: InType[], cb: (val: InType) => OutType) => {
  let newArr = [];
  for (const item of arr) {
  }
  return newArr;
};


demo/js.js
// JavaScript standard built-in objects, along with their methods and properties.
import year from './module-name.js';
import * as p from './module-name.js';
export {p};

const valueRGB = () => {
  return ~~(Math.random() * 256);
};

year.addEventListener('click', function (event) {
  event.preventDefault();
  console.log('click');
  year.style.color = `rgb(${valueRGB()}, ${valueRGB()}, ${valueRGB()})`;
});

const CONSTANT_PI = 3.141592653589793;
const CONSTANT_E = 2.718_281_828_4590;
const regexp = /ab+c/i;
let templateLiteral = `string ${regexp} ${'you'}`;

const myObject = {
  birthYear: 2000,
  garbage: 'jdccnksc',
  age() {
    return 2050 - this.birthYear;
  },
};
delete myObject.garbage;

// prettier-ignore
const array = new Array(globalThis, undefined,
  typeof null, NaN !== NaN, {} !== {}, CONSTANT_PI, true,
// prettier-ignore
  "string", !!'string', password, new Intl.Locale('en'),
  ~~3.4, 10n, -0 != 0, Infinity, functionDeclaration, // hoisted to the top of the enclosing function or global scope. You can use the function before you declared it
  myObject, typeof Symbol, [] instanceof Object, (async () => await this)()
);

const [a1, b1, ...rest] = [...(new Set([1, 3, 4, 1, 3, 1, 5]))];
console.log(a1, b1, rest);
const {birthYear, garbage} = myObject;

for (const iterator of array) {
  console.log(iterator);
}

void array.at(1), array?.at(-4);
try {
  throw new Error('Error message!');
} catch (error) {
  void error;
}
// prettier-ignore
void (
  (
    (
      (
        (
          (
            (((((((""+''+``)))))))
          )
        )
      )
    )
  )
)
[[[[[[[[[[[[]]]]]]]]]]]]
// prettier-ignore
{{{{{{{{{{{{}}}}}}}}}}}}
// Functions and classes
const functionExpression = function (a = 1, c = a ** a) {
  return arguments;
};
function* yieldArray(number = 3) {
  yield* Array.from({length: number}, (_, i) =>
    (i + 1) % (3 * 5) === 0
      ? 'FizzBuzz'
      : (i + 1) % 5 === 0
      ? 'Buzz'
      : (i + 1) % 3 === 0
      ? 'Fizz'
      : i + 1
  );
}
let y = yieldArray(15);
let z;
do {
  z = y.next();
  if (z?.value) console.log(z?.value);
} while (!z?.done);

async function* functionDeclaration(a, c) {
  return a - c;
}
void functionExpression(3);

functionExpression.call(this, 3);
functionExpression.bind(this, 3);
// @ts-check
/**
 * a human
 * @typedef {{birthYear: number}} Human
 */
/**
 * showAge a human age
 * @typedef {function(): string} ShowAge
 */
/**
 * @typedef PublicField
 * @public field (simular to property, available on created object)
 */
/**
 * @typedef PrivateField
 * @private field (not accessible outside of class)
 */
/**
 * class Human
 * @class {Human} human - representing a {@link Human}
 * @abstract
 * Classes just syntactic sugar over constructor functions.
 * Classes aren't hoisted.
 * Classes are first-class citizens.
 * Class body is always executed in 'strict mode'.
 */
class Human {
  /**
   * Create a human.
   * @param {number} birthYear - The birthYear a Number parameter.
   */
  constructor(birthYear) {
    this.birthYear = birthYear;
  }
  /**
   * @private {method} #age - calculate a human age (not accessible outside of class).
   * @return {Number} Number - A human age.
   */
  #age() {
    return 2050 - this.birthYear;
  }
  /**
   * @getter {method} {@link ShowAge} - get the #age value
   * @example <caption>Example usage of showAge.</caption>
   * human.showAge: Number;
   * @return {Number} Number - The #age value
   */
  get showAge() {
    return this.#age();
  }
}

/**
 * Class representing a Person (Child class)
 * @extends Human (Parent class). Inheritance between classes, automatically sets prototype
 */
class Person extends Human {
  /**
   * @public {@link PublicField}
   */
  firstName;
  /**
   * @public {@link PublicField}
   */
  lastName;
  /**
   * @private {@link PrivateField}
   */
  #id = 1001001;
  /**
   * @private {@link PrivateField}
   */
  #creditCardNum = '**** **** **** 1111';
  /**
   * @private {@link PrivateField}
   */
  #pin = 1111;
  /**
   * @static {@link PublicField}
   */
  static car = 'tesla';
  /**
   * Create a person.
   * Constructor method, called by new operator. Mandatory in regular class, might be omitted in a child class.
   * @param {!String} firstName - The firstName a String param.
   * @param {!String} lastName - The lastName a String param.
   * @param {!Number} birthYear - The birthYear a Number param.
   * @param {!Number} pin - The pin a Number param.
   */
  constructor(firstName, lastName, birthYear, pin) {
    /**
     * Create a human.
     * Call to parent (super) class (necessary with extends). Needs to happend before accessing this.
     */
    super(birthYear);
    // Instance property (available on created object)
    this.firstName = firstName;
    this.LastName = lastName;
    // Redefining private field
    this.#pin = pin;
  }
  /**
   * @public method introduce.
   * @return {String} String - introduce and greeting.
   */
  introduce() {
    return `I am ${this.firstName}, my pleasure to meet you in person!}`;
  }
  /**
   * @static public method printId (available on class. Can not access instance properties nor methods, only static ones) .
   * @print String - id.
   */
  static printId() {
    console.log(this.#id);
  }
  /**
   * @static protected method _car (name convention to show other developers to use method in class only) .
   * @return {String} String - key.
   */
  static _car() {
    return `Here are the keys from my ${Person.car}`;
  }
  /**
   * @public method showSecret.
   * @print String - Take Whatever You Want.
   * @example <caption>Example usage of showSecret.</caption>
   * person.showSecret(): String;
   * @abstract Referencing protected method, private method and field
   */
  showSecret() {
    console.log(
      `${Person._car()}, card ${this.#showCard()} and pin ${this.#pin}`
    );
  }
  /**
   * @private method
   * @return {String} Credit Card Number
   */
  #showCard() {
    return this.#creditCardNum;
  }
  /**
   * @setter {method} id - set the persons id value
   * @example <caption>Example usage of id.</caption>
   * person.id(idNum: number): void;
   * @return {Number} Number - The #id value
   * @abstract (use _ to set property with same name as method, and also add getter)
   */
  set id(idNum) {
    this.#id = idNum;
  }
  /**
   * @getter {method} id - get the persons id value
   * @example <caption>Example usage of id.</caption>
   * person.id: Number;
   * @return {Number} Number - The #id value
   */
  get id() {
    return this.#id;
  }
}
// Creating new object with new operator
const person = new Person('Dom', 'N', 2030, 2221);
person.showSecret();

console.log(person.showAge);

switch ('hi') {
  case 'hello':
  case 'hi':
    void 'greeting';
    break;
  default:
    break;
}

while (!true) {
  ('');
}

for (let i = 0; i < array.length; i++) {
  const e = array[i];
  if (!e) continue;
  else {
    void e;
  }
  if (i === 100) break;
}

for (const key in myObject) {
  if (myObject.hasOwnProperty.call(myObject, key)) {
    const element = myObject[key];
    void element;
  }
}
// rettier-ignore Assignment Operators
void = *= /= %= += -= <<= >>= >>>= &= ^= |= **= &&= ||= ??=
// prettier-ignore Arithmetic Operators
void ++ -- + - ** * / %
// prettier-ignore Relational Operators
void <, >, <=, >=
// prettier-ignore Comparison operators
void ==  !=  ===  !== 
// prettier-ignore Bitwise Operators
void ~ << >> >>> & | ^
// Binary logical operators
void && ! || ??
